src/main/java/org/minions/devfund/carlosorellana/Squarelotron.java

```diff
@@ -0,0 +1,254 @@
+package org.minions.devfund.carlosorellana;
+
+/**
+ * The goal is execute the Squarelotron methods.
+ *
+ * @author Carlos Orellana.
+ * @since 07/17/2018.
+ */
+public class Squarelotron {
+
+    private int[][] squarelotron;
+    private int size;
+
+    /**
+     * Gets Squarelotron Matrix.
+     *
+     * @return Squarelotron Matrix.
+     */
+    public int[][] getSquarelotron() {
+        return squarelotron.clone();
+    }
+
+    /**
+     * Constructor Sqarelotron.
+     *
+     * @param n - Size of squarelotron array.
+     */
+    public Squarelotron(final int n) {
+        this.size = n;
+        this.squarelotron = new int[size][size];
+
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                squarelotron[i][j] = size * i + j + 1;
+            }
+        }
+    }
+
+    /**
+     * Gets the maximum number of ring in the squarelotron.
+     *
+     * @return Numbers maximum of rings.
+     */
+    public int getMaxRings() {
+        return (this.size % 2 == 0) ? size / 2 : size / 2 + 1;
+    }
+
+    /**
+     * Modifies A Upside-Down Flip of the outer ring of Squarelotron.
+     *
+     * @param ring - Number of ring.
+     * @return Instance new Squarelotron.
+     */
+    public Squarelotron upsideDownFlip(final int ring) {
+
+        if (ring > this.getMaxRings() || ring < 1) {
+            throw new NumberFormatException();
+        }
+
+        Squarelotron squarelotronTemporal = new Squarelotron(this.size);
+
+        int first = ring - 1;
+        int last = size - ring;
+
+        for (int i = 0; i <= size - 1; i++) {
+            for (int j = 0; j <= size - 1; j++) {
+
+                squarelotronTemporal = createNewMatrixToUpsideDownFlip(i, j, first, last, squarelotronTemporal);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment; "squarelotronTemporal" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4165)